### DISCUSSION TITLE
GUI for /listclaims with sorting.

### DISCUSSION THREAD

**Discussion by TomLewis** [GriefPrevention/GriefPrevention#990] [Ideas]
I have always had issues with players struggling with `/claimslist` being a not very user friendly.
Really easy solution, have it open a GUI with items as each claim to organize claims.
Example from another plugin on how they could be shown.

![2020-08-27_20 10 57](https://user-images.githubusercontent.com/144046/91581473-603fd500-e946-11ea-8891-e7079f3bf107.png)

Other features could include:

- Pagination for multiple pages
- Being able to set what block you want to make the GUI pretty. Defaults could be the lower left corner of a claim that the block shown is shown in the GUI.
- Option to sort claims from a top menu of clickable items.
  - Sort by date created
  - Sort by size of claim
  - Sort by most trusts
  - Sort by name (allow naming them from the GUI)
- Name plots in the GUI by Middle Clicking/Shift+right click or something, this could then open the Anvil GUI to enter a name (Like dtltraders does for their shop GUI naming items)
- Have a key combo to delete like shift+middle click or something. 
- Have a quick way to create a claim around you from a GUI click.
- Clicking a claim opens up a new UI that lets you mange that claim
  - Being able to list all Trusted members, which shows all their heads in the GUI
  - Permission based options are shown in here, like admins will have a remote delete claim for example.

Expandable options later:
- Standing in a claim and opening up the GUI will open up a GUI of options for that specific claim, for example requesting to an admin it be deleted

Please take more inspiration from:
https://www.spigotmc.org/resources/lands-minecraft-land-claim-plugin-grief-prevention-protection-gui-management-wars-1-16-support.53313/

**Comment by RoboMWM** [comment]
If anyone's up for doing this, you may also want to consider book GUIs, which are less buggier and better suited for text-based information (while supporting the same click and hover events as in chat).
